radixdlt/src/integration/java/com/radixdlt/consensus/MultiNodeConsensusTest.java

```diff
@@ -0,0 +1,191 @@
+import com.radixdlt.consensus.liveness.RotatingLeaders;
+import com.radixdlt.consensus.liveness.Dictatorship;
+import com.radixdlt.consensus.safety.QuorumRequirements;
+import com.radixdlt.consensus.safety.SafetyRules;
+import com.radixdlt.consensus.safety.SafetyState;
+import com.radixdlt.consensus.safety.WhitelistQuorum;
+import com.radixdlt.crypto.ECDSASignatures;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.engine.RadixEngine;
+import com.radixdlt.mempool.Mempool;
+import com.radixdlt.network.TestEventCoordinatorNetwork;
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.observers.TestObserver;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import org.apache.commons.math3.util.Pair;
+import org.junit.Before;
+import org.junit.Test;
+import org.radix.logging.Logger;
+import org.radix.logging.Logging;
+
+public class MultiNodeConsensusTest {
+	private static final Logger log = Logging.getLogger("Test");
+	private final TestEventCoordinatorNetwork testEventCoordinatorNetwork = new TestEventCoordinatorNetwork();
+
+	private Atom genesis;
+	private Vertex genesisVertex;
+	private QuorumCertificate genesisQC;
+
+	@Before
+	public void setup() {
+		this.genesis = mock(Atom.class);
+		when(this.genesis.toString()).thenReturn(Long.toHexString(0));
+		this.genesisVertex = Vertex.createGenesis(genesis);
+		this.genesisQC = new QuorumCertificate(
+			new VertexMetadata(View.genesis(), genesisVertex.getId(), null, null),
+			new ECDSASignatures()
+		);
+	}
+
+	private ChainedBFT createBFTInstance(
+		ECKeyPair key,
+		ProposerElection proposerElection,
+		QuorumRequirements quorumRequirements,
+		VertexStore vertexStore
+	) {
+		Mempool mempool = mock(Mempool.class);
+		AtomicLong atomId = new AtomicLong();
+		doAnswer(inv -> {
+			Atom atom = mock(Atom.class);
+			when(atom.toString()).thenReturn(Long.toHexString(atomId.incrementAndGet()));
+			return Collections.singletonList(atom);
+		}).when(mempool).getAtoms(anyInt(), anySet());
+		ProposalGenerator proposalGenerator = new ProposalGenerator(vertexStore, mempool);
+		SafetyRules safetyRules = new SafetyRules(key, vertexStore, SafetyState.initialState());
+		PacemakerImpl pacemaker = new PacemakerImpl(quorumRequirements, Executors.newSingleThreadScheduledExecutor());
+		PendingVotes pendingVotes = new PendingVotes(quorumRequirements);
+		EventCoordinator eventCoordinator = new EventCoordinator(
+			proposalGenerator,
+			mempool,
+			testEventCoordinatorNetwork.getNetworkSender(key.getUID()),
+			safetyRules,
+			pacemaker,
+			vertexStore,
+			pendingVotes,
+			proposerElection,
+			key
+		);
+
+		return new ChainedBFT(
+			eventCoordinator,
+			testEventCoordinatorNetwork.getNetworkRx(key.getUID()),
+			pacemaker
+		);
+	}
+
+	private List<TestObserver<Vertex>> runBFT(
```

Comment:
Even though this is private it would be helpful to explain what this returns (the stream of committed vertices)